src/Model/User.php

```diff
@@ -78,21 +78,105 @@ public function __construct( \WP_User $user ) {
 	 */
 	protected function is_private() {
 
+		/**
+		 * Filter whether the user is private.
+		 *
+		 * If true, the user will be considered private without further checks.
+		 *
+		 * @param mixed|null|boolean $is_private If set, return without executing the manual query
+		 *                                       to see if the user has published posts.
+		 * @param User               $this       Instance of the User Model
+		 */
+		$is_private = apply_filters( 'graphql_user_model_is_private', null, $this );
+
+		/**
+		 * If the filter returns true, we can return now as the user is considered private.
+		 *
+		 * Otherwise we need to continue and check if the User should be considered private or not
+		 */
+		if ( null !== $is_private ) {
+			return (bool) $is_private;
+		}
+
+		/**
+		 * If the requesting user does not have "list_users" capabilities and the current user is not
+		 * the user being requested, we need to check if the user being requested has published posts
+		 * which makes them a non-private entity.
+		 */
 		if ( ! current_user_can( 'list_users' ) && false === $this->owner_matches_current_user() ) {
 
 			/**
-			 * @todo: We should handle this check in a Deferred resolver. Right now it queries once per user
-			 *      but we _could_ query once for _all_ users.
-			 *
-			 *      For now, we only query if the current user doesn't have list_users, instead of querying
-			 *      for ALL users. Slightly more efficient for authenticated users at least.
+			 * Get allowed Post Types
+			 */
+			$post_types = \WPGraphQL::get_allowed_post_types();
+			unset( $post_types['revision'] );
+			unset( $post_types['attachment'] );
```

Comment:
I can understand revisions, but why unset attachments? I think it might be better to do an array intersect on all public post types maybe?

This gets pretty complicated actually, as we _technically_ shouldn't expose any authors that don't have published posts exposed in GraphQL. So if someone goes and locks down the `Pages` type, and a user only has posts in that post type, we technically shouldn't expose them. In a headless context, there would be no other way to enumerate that user. 🤔

Reply:
Attachments don't get the `publish` status, just inherit. 

So, it's a waste of energy to check for `publish` attachments, as there won't be any.

Reply:
ah, I see. Would be good to add an inline comment about this for context, since this seems super arbitrary.